src/TSHelper.ts

```diff
@@ -80,6 +80,22 @@ export class TSHelper {
             || (ts.isBinaryExpression(node.parent) && ts.isArrayLiteralExpression(node.parent.left)));
     }
 
+    // iterate over a type and its bases until the callback returns true.
+    public static forAllTypes(type: ts.Type, callback: (type: ts.Type) => boolean): boolean {
```

Comment:
slightly missleading name, I think 'forTypeOrAnySupertype' would be clearer

Reply:
agreed.